example/integration_test/viewport_test.dart

```diff
@@ -0,0 +1,133 @@
+    // apply default viewport
+    app.main(viewport: StyleDefaultViewportState());
+    await tester.pumpAndSettle();
+    final cameraAfterViewport = await mapboxMap.getCameraState();
+
+    // check that the came is reset to the default viewport
+    expect(
+        cameraAfterViewport.center.coordinates.lng,
+        moreOrLessEquals(
+            styleDefaultCamera.center!.coordinates.lng.toDouble()));
+    expect(
+        cameraAfterViewport.center.coordinates.lat,
+        moreOrLessEquals(
+            styleDefaultCamera.center!.coordinates.lat.toDouble()));
+    expect(cameraAfterViewport.zoom,
+        moreOrLessEquals(styleDefaultCamera.zoom!.toDouble()));
+    expect(cameraAfterViewport.bearing,
+        moreOrLessEquals(styleDefaultCamera.bearing!.toDouble()));
+    expect(cameraAfterViewport.pitch,
+        moreOrLessEquals(styleDefaultCamera.pitch!.toDouble()));
+  });
+
+  testWidgets('OverviewViewportState test', (WidgetTester tester) async {
+    final overviewViewport = OverviewViewportState(
+      geometry: Polygon.fromPoints(points: [
+        [
+          Point(coordinates: Position(0, 0)),
+          Point(coordinates: Position(0, 10)),
+          Point(coordinates: Position(10, 10)),
+          Point(coordinates: Position(10, 0)),
+          Point(coordinates: Position(0, 0))
+        ]
+      ]),
+      bearing: 30.0,
+      pitch: 45.0,
+    );
+    final mapFuture = app.main(viewport: overviewViewport);
+    await tester.pumpAndSettle();
+    final mapboxMap = await mapFuture;
+    final camera = await mapboxMap.getCameraState();
+
+    // Android has a bug that overview viewport state is not applying correctly.
+    // The bug appears rarely in real use case scenarios, but is more prominent in tests.
+    // TODO: Remove this once https://mapbox.atlassian.net/browse/MAPSAND-1907 is fixed.
+    if (!Platform.isAndroid) {
+      expect(camera.center.coordinates.lng, moreOrLessEquals(5, epsilon: 0.1));
+      expect(camera.center.coordinates.lat, moreOrLessEquals(5, epsilon: 0.1));
+    }
+    expect(
+        camera.bearing, moreOrLessEquals(overviewViewport.bearing!.toDouble()));
+    expect(camera.pitch, moreOrLessEquals(overviewViewport.pitch!.toDouble()));
+  });
+
+  testWidgets('IdleViewportState test', (WidgetTester tester) async {
+    final idleViewport = IdleViewportState();
+    final mapFuture = app.main(viewport: idleViewport);
+    await tester.pumpAndSettle();
+    await mapFuture;
+
+    // no-op viewport state, camera should not change
+  });
+
+  testWidgets('FollowPuckViewportState test', (WidgetTester tester) async {
+    final followPuckViewport = FollowPuckViewportState(
+      zoom: 16.0,
+      bearing: FollowPuckViewportStateBearingConstant(30.0),
+      pitch: 45.0,
+    );
+    final mapFuture = app.main(viewport: followPuckViewport);
+    await tester.pumpAndSettle();
+    final _ = await mapFuture;
+
+    // TODO: test that the camera is updated when custom location provider is supported
```

Comment:
Create a ticket for this?

Reply:
The ticket is [here](https://mapbox.atlassian.net/browse/MAPSFLT-274)